Source/WebKit/UIProcess/API/glib/WebKitCookieManager.cpp

```diff
@@ -517,3 +517,106 @@ void webkit_cookie_manager_delete_all_cookies(WebKitCookieManager* manager)
     webkit_website_data_manager_clear(manager->priv->dataManager, WEBKIT_WEBSITE_DATA_COOKIES, 0, nullptr, nullptr, nullptr);
 }
 #endif
+
+/**
+ * webkit_cookie_manager_set_all_cookies:
+ * @cookie_manager: a #WebKitCookieManager
+ * @cookies: (element-type SoupCookie) (transfer none): a #GList of #SoupCookie to be added
+ * @operation: a #WebKitCookieManagerOperation
+ * @cancellable: (allow-none): a #GCancellable or %NULL to ignore
+ * @callback: (scope async): a #GAsyncReadyCallback to call when the request is satisfied
+ * @user_data: (closure): the data to pass to callback function
+ *
+ * Asynchronously set a list of #SoupCookie to the underlying storage.
+ *
+ * When the operation is finished, @callback will be called. You can then call
+ * webkit_cookie_manager_set_all_cookies_finish() to get the result of the operation.
+ *
+ * Since: 2.42
+ */
+void webkit_cookie_manager_set_all_cookies(WebKitCookieManager* manager, GList* cookies, WebKitCookieManagerOperation operation, GCancellable* cancellable, GAsyncReadyCallback callback, gpointer userData)
+{
+    g_return_if_fail(WEBKIT_IS_COOKIE_MANAGER(manager));
```

Comment:
Do we allow to pass null for cookies? I guess we don't because we can `delete_all_cookies` for that.

Reply:
And we should not allow to pass empty list of cookies - implicit removing all cookies, right?

Reply:
Yes, we already have `delete_all_cookies` for that